lib/serial_terminal.pm

```diff
@@ -367,6 +367,12 @@ sub select_serial_terminal {
     my $backend = get_required_var('BACKEND');
     my $console;
 
+    if (get_var('OFW') && !check_var('OFW_HVC_SERIAL_CONSOLE_ADDED', '1')) {
+        select_console 'root-console';
+        prepare_serial_console;
+        set_var('OFW_HVC_SERIAL_CONSOLE_ADDED', 1);
+    }
```

Comment:
I'm ok with this, but wouldn't be better to add the guarder to `prepare_serial_console()`?

https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c40110c3ef1ef43138c7547ad0a3ecb50eaf22d7/lib/serial_terminal.pm#L71-L78

BTW adding `prepare_serial_console()` would mean that the other calls below could be deleted, right?
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c40110c3ef1ef43138c7547ad0a3ecb50eaf22d7/tests/console/system_prepare.pm#L33-L38
https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/c40110c3ef1ef43138c7547ad0a3ecb50eaf22d7/tests/shutdown/cleanup_before_shutdown.pm#L50-L56

Reply:
I think only ppc64le does require to start hvc consoles. I will remove other `prepare_serial_console`

Reply:
I suppose code in `prepare_serial_console()` should be correct (feel free to retest it and suggest change):
* `hvc0` is needed only for SLE < 12-SP2 on all but s390x. NOTE: we still test SLE11-SP4 on kernel (it should be kept)
* `hvc1` is needed for ppc64le. Maybe `get_var('OFW')` could be replaced by `is_ppc64le`, but not sure
* `hvc1` is also needed for other archs except ppc64le for `user-virtio-terminal`
* `hvc2` is needed for ppc64le for `user-virtio-terminal`.
But IMHO 1) call `prepare_serial_console()` only in `prepare_serial_console()` and guard is just by new variable `SERIAL_CONSOLE_ADDED` (it should be for any arch) 2) remove all `prepare_serial_console()` calls elsewhere.

Reply:
I'd hope that all code uses `select_serial_terminal()` or `select_user_serial_terminal()` so it should work, but maybe I'm missing something. But sure, postpone this change to separate effort.

Reply:
I've read that before, but that expects all different installer (SLES / Tumbleweed / JeOS, ...) need to schedule `tests/installation/first_boot.pm` (I know they will need it for different things anyway). Also we have preinstalled qcow for old SLES versions (where installer is not tested any more), therefore this setup must be done manually. Having required config in `select_serial_terminal()` would not need it.

Reply:
Or we can move `console/system_prepare.pm` immediately after `installation/*` modules. That'll also make serial terminal available for the rest of post-install setup.

Reply:
`system_prepare` was not scheduled in qam-minimal tests, I will keep it as it is and just schedule `system_prepare`.
If it would be only about hvc console on ppc64 then I would do it within select_serial_terminal, but it is adding also user console, so I think it's good as it is, done in `system_prepare` and after installation in `cleanup_before_shutdown`